### PR TITLE
[FIX] web: avoid expensive 404 during js tests

### PR DESCRIPTION
**Pull request by Xavier-Do**
A 404 will take 3 to 5 seconds to be resolved and execute +- 1900 query, to return
a rendered page which is quite expensive, especially when multiple missing images
are rendered in a view.

Catching static route and marking them as not frontend will help to avoid to handle miss
on static resources. In this case server returns a standard 404.

This commit also fix a iframe src in order to avoid a 404 on
/web/(test )/report/html/some_report (thx to aab-odoo)

### CODE REVIEW

**Comment by odony** [comment]
I think this could break fallback routing of attachments via `serve_attachment()`, which could have URLs that match `/<module>/static`, typically used by data-only modules.

As the `handle_exception()` logic already distinguishes "frontend requests" from "other requests", and serves a lightweight response to the latter, we might want to look instead at this piece of code: 
https://github.com/odoo/odoo/blob/master/addons/http_routing/models/ir_http.py#L428-L432

Switching `is_frontend` to false when the request path has the form `/<module>/static/*` might reduce a lot the number of useless work performed for a 404 in static resources.

/cc @xmo-odoo

**Comment by Xavier-Do** [comment]
@odony 9 queries with this version, good enough for me.

File: addons/http_routing/models/ir_http.py [hunk 1]
```diff
@@ -427,8 +427,12 @@ def _dispatch(cls):
             request.is_frontend = func.routing.get('website', False)
         except werkzeug.exceptions.NotFound as e:
             # either we have a language prefixed route, either a real 404
-            # in all cases, website processes them
-            request.is_frontend = True
+            # in all cases, website processes them exept if second element is static
+            # Checking static will avoid to generate an expensive 404 web page since
+            # most of the time the browser is loading and inexisting assets or image. A standard 404 is enough.
+            # Earlier check would be difficult since we don't want to break data modules
+            split = request.httprequest.path.split('/')[1]
```
Comment (odony) [inline addons/http_routing/models/ir_http.py]: You probably meant this, right?
`split` could also have a more descriptive name, e.g. `path_components`, but at this point we're nit-picking ;-)

```suggestion
            split = request.httprequest.path.split('/')
```

**Comment by odony** [comment]
> @odony 9 queries with this version, good enough for me.

Cool :+1: 

As this impacts the website routing, I'd ask @JKE-be or @rdeodoo to have a quick look here as well.

File: addons/http_routing/models/ir_http.py [hunk 2]
```diff
@@ -427,8 +427,12 @@ def _dispatch(cls):
...
         except werkzeug.exceptions.NotFound as e:
             # either we have a language prefixed route, either a real 404
-            # in all cases, website processes them
-            request.is_frontend = True
+            # in all cases, website processes them exept if second element is static
+            # Checking static will avoid to generate an expensive 404 web page since
+            # most of the time the browser is loading and inexisting assets or image. A standard 404 is enough.
+            # Earlier check would be difficult since we don't want to break data modules
+            split = request.httprequest.path.split('/')
+            request.is_frontend = len(split) < 2 or split[1] != 'static'
```
Comment (rdeodoo) [inline addons/http_routing/models/ir_http.py]: If this is for `/module_name/static/whetever`, `split[1]` will be `module_name` and not `static` no?

Comment (Xavier-Do) [reply addons/http_routing/models/ir_http.py]: Obviously I made some quick assumption that it was working when it wasn't. New version pushed, thanks for the help.

Comment (rdeodoo) [reply addons/http_routing/models/ir_http.py]: Always happy to help, especially if it is website related ;-)

**Comment by Xavier-Do** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**